modules/cudafilters/test/test_filters.cpp

```diff
@@ -647,4 +648,57 @@ INSTANTIATE_TEST_CASE_P(CUDA_Filters, MorphEx, testing::Combine(
     testing::Values(Iterations(1), Iterations(2), Iterations(3)),
     WHOLE_SUBMAT));
 
+/////////////////////////////////////////////////////////////////////////////////////////////////
+// Median
+
+
+PARAM_TEST_CASE(Median, cv::cuda::DeviceInfo, cv::Size, MatDepth,  KernelSize, UseRoi)
+{
+    cv::cuda::DeviceInfo devInfo;
+    cv::Size size;
+    int type;
+    int kernel;
+    bool useRoi;
+
+    virtual void SetUp()
+    {
+        devInfo = GET_PARAM(0);
+        size = GET_PARAM(1);
+        type = GET_PARAM(2);
+        kernel = GET_PARAM(3);
+        useRoi = GET_PARAM(4);
+
+        cv::cuda::setDevice(devInfo.deviceID());
+    }
+};
+
+CUDA_TEST_P(Median, Accuracy)
+{
+    cv::Mat src = randomMat(size, type);
+
+    cv::Ptr<cv::cuda::Filter> median = cv::cuda::createMedianFilter(src.type(), kernel);
+
+    cv::cuda::GpuMat dst = createMat(size, type, useRoi);
+    median->apply(loadMat(src, useRoi), dst);
+
+    cv::Mat dst_gold;
+    cv::medianBlur(src,dst_gold,kernel);
+
+    EXPECT_MAT_SIMILAR(dst_gold, dst, 1.0);
```

Comment:
Unfortunately, I don't see border mode specification in the documentation of the original CPU version. But something should be assumed in the any specific implementation, like `BORDER_REPLICATE`/`BORDER_REFLECT`/`BORDER_REFLECT101` (but `BORDER_WRAP` might provide completely different results).
To completely "ignore" border policy (accept any implementation) you can use this [approach](https://github.com/Itseez/opencv/blob/3.1.0/modules/cudafilters/test/test_filters.cpp#L532):
```
EXPECT_MAT_NEAR(getInnerROI(dst_gold, ksize), getInnerROI(dst, ksize), 0.0);
```
where use `ksize` or `ksize - 1` (not sure about `getInnerROI` function implementation).